src/gpu-compute/shader.cc

```diff
@@ -221,6 +221,11 @@ Shader::prepareInvalidate(HSAQueueEntry *task) {
         // all necessary INV flags are all set now, call cu to execute
         cuList[i_cu]->doInvalidate(req, task->dispatchId());
 
+
+        if ((i_cu % 4) == 0) {
```

Comment:
Do we have a variable in the Python for this?  4 is a magic number because of how many CUs share an SQC.  To make this code more resilient we should really add/use such a variable to avoid problems later.

Reply:
We have one in the top level config script but did not in the Shader/Compute Unit models. I added one and use it in this logic instead. It gets its value from the top level config script. I wanted to do this in my initial commit but completely forgot about it. Thanks for pointing it out.

Reply:
Probably we'd need to update that script too to reflect this change, but it shouldn't be part of this patch